Ruby/lib/algorithms.rb

```diff
@@ -4,17 +4,41 @@
 # The result is called a 'digital root'.
 # Do not use string conversion within your method.
 def digital_root(number)
-
+  # sum = 0
+  number = number.to_s.split('').map(&:to_i)
+  sum = number.reduce(:+) #{|sum, num| sum + num}
+  # sum #17
+  sum >= 10 ? digital_root(sum) : sum
+  # if sum >=10
+  #   digital_root(sum)
+  # else
+  #   sum
+  # end
 end
 
 # Write a function that takes a message and an increment amount.
 # Output the same letters shifted by that amount in the alphabet.
 # Assume lowercase and no punctuation.
 # Preserve spaces.
-def caesar_cipher(string, shift)
+def caesar_cypher(string, shift)
+
+ p num_string = string.chars.map(&:ord) #changes number into array of the numbers [104, 105, 32, 104, 105, 32]
+  p "*****"
+
+ p encrypted_num = num_string.map {|c| c + shift} #shifts the numbers according to the argument passed [115, 116, 43, 115, 116, 43]
+ p "*****"
+  num_string.each do |num|
+    if num < 97
+    num_string.gsub!('num', ' ')
+    end
+  end
+ p encrypted_let = encrypted_num.map {|c| c.chr}.join #change into letters and join  "st+st+"
+ p "*****"
```

Comment:
The specs are actually expecting a string to be returned, not printed to the terminal. Once that's fixed, this will work except when the string contains one of more spaces. Would the instructions be more clear if they said "Return the same letters shifted by that amount in the alphabet." instead of "Output the same letters shifted by that amount in the alphabet."?